### ISSUE TITLE
right-click and middle-click on links triggers swup navigation in Firefox

### ISSUE THREAD

**Issue by amrnn** [gmrchk/swup#72]
When using Firefox if I right-click or middle-click a link, swup immediately navigates to that link.

**Comment by drivebass** [comment]
Same issue here

**Comment by gmrchk** [comment]
This is a pretty interesting one. It seems that guys at Firefox decided that [specs](https://www.w3.org/TR/uievents/#event-type-click) are not specific in what should trigger the click event, or at least that's what I understood from [here](https://bugzilla.mozilla.org/show_bug.cgi?id=71705). I'm not a native English speaker, but it seems to me that specs are pretty clear about it. 

> The click event should only be fired for the primary pointer	button (i.e., when button value is 0, buttons value is 1). Secondary buttons (like the middle or right button on a standard mouse) MUST NOT fire click events.  

It also seems that FF triggers click in different situations when the listener is attached to document, which is discussed [here](https://stackoverflow.com/questions/43144995/mouse-right-click-on-firefox-triggers-click-event).

To summarize, I think this is FF issue. What do you guys think?

In any way, we can't wait for browsers to unify this, so fixed in latest version using `event.button`.

**Comment by drivebass** [comment]
Yes it's definitely FF's issue, confusing implementation of the event's spec. It works great with the fix!

**Comment by amrnn** [comment]
Thanks, works well with the fix.
